src/manifest.c

```diff
@@ -474,35 +474,31 @@ gboolean check_manifest_internal(const RaucManifest *mf, GError **error)
 	switch (mf->bundle_format) {
 		case R_MANIFEST_FORMAT_PLAIN:
 			break; /* no additional data needed */
-		case R_MANIFEST_FORMAT_CRYPT: {
-			if (mf->bundle_crypt_key) {
-				g_set_error_literal(error, R_MANIFEST_ERROR, R_MANIFEST_CHECK_ERROR, "Unexpected key for crypt bundle in internal manifest");
-				goto out;
-			}
-		};
-		/* Fallthrough */
-		case R_MANIFEST_FORMAT_VERITY: {
-			if (mf->bundle_verity_hash) {
-				g_set_error(error, R_MANIFEST_ERROR, R_MANIFEST_CHECK_ERROR, "Unexpected hash for %s bundle in internal manifest", r_manifest_bundle_format_to_str(mf->bundle_format));
-				goto out;
-			}
-			if (mf->bundle_verity_salt) {
-				g_set_error(error, R_MANIFEST_ERROR, R_MANIFEST_CHECK_ERROR, "Unexpected hash for %s bundle in internal manifest", r_manifest_bundle_format_to_str(mf->bundle_format));
-				goto out;
-			}
-			if (mf->bundle_verity_size) {
-				g_set_error(error, R_MANIFEST_ERROR, R_MANIFEST_CHECK_ERROR, "Unexpected hash for %s bundle in internal manifest", r_manifest_bundle_format_to_str(mf->bundle_format));
-				goto out;
-			}
-
-			break;
-		};
+		case R_MANIFEST_FORMAT_CRYPT:
+		case R_MANIFEST_FORMAT_VERITY:
 		default: {
 			g_set_error(error, R_MANIFEST_ERROR, R_MANIFEST_CHECK_ERROR, "Unsupported bundle format");
```

Comment:
Not sure if this message isn't too broad when used as a fallthrough. However, we should not run into this case.
An alternative would be to add something like

``` c
g_set_error(error, R_MANIFEST_ERROR, R_MANIFEST_CHECK_ERROR, "Bundle format '%s' not allowed for internal manifest", r_manifest_bundle_format_to_str(mf->bundle_format));
```

above.